async_processor/worker.py

```diff
@@ -27,6 +29,50 @@
     from async_processor.processor import AsyncProcessorWrapper
 
 
+async def _retry_with_exponential_backoff(
+    coroutine: Callable,
+    kwargs: Optional[Dict] = None,
+    num_retries: int = 5,
+    # base_ms needs to be more than 1. sub-ms does not work.
+    base_ms: int = 5,
+    jitter_ms: Optional[Tuple[int, int]] = (0, 100),
+    max_wait_ms: float = 2000,
+    retry_exceptions=(Exception,),
+) -> Any:
+    retry_count = 0
+    exceptions_raised = []
+
+    kwargs = kwargs or {}
+
+    execution_name = getattr(coroutine, "__name__", "NO_NAME")
+    execution_name = (
+        f"{execution_name}-{''.join(random.choices(string.ascii_letters, k=4))}"
+    )
+
+    while retry_count <= num_retries:
+        try:
+            return await coroutine(**kwargs)
+        except retry_exceptions as ex:
+            exceptions_raised.append(ex)
+            retry_count += 1
+            sleep_ms = base_ms**retry_count
```

Comment:
Isn't this too exponential? maybe we should have something slower growing or even linear backoff `base_ms * retry_count` ?

Reply:
oh never mind, I see we have `max_wait_ms`